### PR TITLE
Document Validator API CLI options

### LINKED ISSUE #336: Doc validator-api options
Document `validator-api` options. See https://github.com/ConsenSys/teku/pull/4890.

### PR DESCRIPTION
**Pull request by rolandtyler**
## Describe the change
Document the new validator API command line options.

## Issue fixed
Fixes #336 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

### CODE REVIEW

File: docs/Reference/Rest_API/Rest.md [hunk 1]
```diff
@@ -61,6 +61,23 @@ You can also use tools such as [Postman] or [cURL] to interact with Teku APIs.
         }
         ```
 
+### Enable the validator REST API
+
+The [validator REST API](https://consensys.github.io/teku/#tag/Validator-Required-Api) is a subset of API methods that are enabled separately from the other REST APIs.
```
Comment (ajsutton) [inline docs/Reference/Rest_API/Rest.md:66]: I'm not sure it's accurate to say they're a subset - they're a completely distinct set of APIs.  The key difference being between REST APIs for the beacon node and these APIs for the validator client.

File: docs/Reference/Rest_API/Rest.md [hunk 2]
```diff
@@ -61,6 +61,23 @@ You can also use tools such as [Postman] or [cURL] to interact with Teku APIs.
         }
         ```
 
+## Enable the validator client API
+
+The [validator client API methods](https://consensys.github.io/teku/#tag/Validator) are enabled separately from the REST API methods.
```
Comment (rolfyone) [inline docs/Reference/Rest_API/Rest.md:66]: this is actually the keymanager api that is enabled on the VC, or potentially the BN. Hopefully at some point should be available at https://consensys.github.io/teku/keymanager-APIs/
@NicolasMassart was looking at publishing to there...

External standard reference is: https://ethereum.github.io/keymanager-APIs/

File: docs/Reference/Rest_API/Rest.md [hunk 3]
```diff
@@ -61,6 +61,23 @@ You can also use tools such as [Postman] or [cURL] to interact with Teku APIs.
...
+
+Enable the validator client API service from the command line by setting the
+[`--validator-api-enabled`](../CLI/CLI-Syntax.md#validator-api-enabled) command line option to `true`.
+
+You can also interact with APIs using Swagger UI by enabling the API documentation endpoint by setting the
+[`--validator-api-docs-enabled`](../CLI/CLI-Syntax.md#validator-api-docs-enabled) option to `true`.
+
+Access the APIs at `http:<interface>:<port>/swagger-ui` where:
```
Comment (rolfyone) [inline docs/Reference/Rest_API/Rest.md:74]: duplicate of above comment, but no swagger-ui.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 4]
```diff
@@ -1507,6 +1507,263 @@ The default is 5051.
...
+
+    ```bash
+    validator-api-docs-enabled: true
+    ```
+
+Set to `true` to enable the Validator REST API documentation.
+The default is `false`.
+
+The documentation can be accessed at `http://<interface>:<port>/swagger-ui` where:
+
```
[the remark this answers is not in the document]
Comment (rolandtyler) [reply docs/Reference/CLI/CLI-Syntax.md:1580]: @rolfyone, for the 3 CLI options related to swagger-ui, `validator-api-docs-enabled`, `validator-api-interface`, and `validator-api-port`; is taking out the swagger-ui comment enough, or should I take those 3 options out of the docs for now as well?  Is there an alternative way to use `validator-api-docs-enabled` besides swagger-ui?

[the remark this answers is not in the document]
Comment (rolfyone) [reply docs/Reference/CLI/CLI-Syntax.md:1637]: validator-api-docs-enabled is generating the openapi spec, so its helpful, but it doesn't enable a web interface...
The `/swagger-docs` endpoint can be used for api generation, so it's worth documenting that it's there, we just shouldn't be mentioning the `swagger-ui`
so if you have docs-enabled, the openapi documentation will be generated to `/swagger-docs` i guess is the way to put it?

File: docs/Reference/Rest_API/Rest.md [hunk 5]
```diff
@@ -68,15 +68,15 @@ The [validator client API methods](https://ethereum.github.io/keymanager-APIs/)
 Enable the validator client API service from the command line by setting the
 [`--validator-api-enabled`](../CLI/CLI-Syntax.md#validator-api-enabled) command line option to `true`.
 
-You can also interact with APIs using Swagger UI by enabling the API documentation endpoint by setting the
+You can also interact with APIs by enabling the API documentation endpoint by setting the
 [`--validator-api-docs-enabled`](../CLI/CLI-Syntax.md#validator-api-docs-enabled) option to `true`.
 
-Access the APIs at `http:<interface>:<port>/swagger-ui` where:
+Access the APIs at `http:<interface>:<port>/swagger-docs` where:
```
Comment (rolfyone) [inline docs/Reference/Rest_API/Rest.md:74]: So the documentation for the APIs is available at swagger-docs, but its a huge mess of JSON that can really only be consumed by something...
`The openapi json document can be consumed from /swagger-docs if --validator-open-api-docs-enabled`?

File: docs/Reference/Rest_API/Rest.md [hunk 6]
```diff
@@ -68,16 +68,16 @@ The [validator client API methods](https://ethereum.github.io/keymanager-APIs/)
 Enable the validator client API service from the command line by setting the
 [`--validator-api-enabled`](../CLI/CLI-Syntax.md#validator-api-enabled) command line option to `true`.
 
-You can also interact with APIs by enabling the API documentation endpoint by setting the
-[`--validator-api-docs-enabled`](../CLI/CLI-Syntax.md#validator-api-docs-enabled) option to `true`.
+Alternatively, the [OpenAPI specifications](https://swagger.io/specification/) for the validator client API can be consumed by an interface such as [swagger-docs](https://swagger.io/docs/).
+In order to generate the specifications, you must first enable the API documentation endpoint.
```
Comment (rolfyone) [inline docs/Reference/Rest_API/Rest.md:72]: The [OpenAPI specifications](https://swagger.io/specification/) for the validator client API are available at `/swagger-docs` when `--validator-api-docs-enabled` is `true`.
The `/swagger-docs` endpoint will define the API if code generators are in use.

Basically removed the use of 'alternatively', Im not sure it fits well, and tried to better explain why you'd enable the endpoint

File: docs/Reference/CLI/CLI-Syntax.md [hunk 7]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
 
 Displays the version and exits.
 
+### validator-api-cors-origins
+
+=== "Syntax"
+
+    ```bash
+    --validator-api-cors-origins[=<url>[,<url>...]...] or "*"
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1629]:
```suggestion
    --validator-api-cors-origins="<URL>"[,"<URL>",...] or "*"
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 8]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-cors-origins: ["http://medomain.com","https://meotherdomain.com"]
+    ```
+
+A list of domain URLs for CORS validation. You must enclose the URLs in double quotes and separate
+them with commas.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1651]:
```suggestion
A comma-separated list of domain URLs for CORS validation.
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 9]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    TEKU_VALIDATOR_API_DOCS_ENABLED=true
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-docs-enabled: true
+    ```
+
+Set to `true` to enable the Validator REST API documentation.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1690]:
```suggestion
Set to `true` to enable the [validator REST API documentation](../Rest_API/Rest.md#enable-the-validator-client-api).
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 10]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+The default is `false`.
+
+When enabling the API documentation endpoint, you must also specify:
+
+* `interface` by using [`--validator-api-interface`](#validator-api-interface)
+* `port` by using [`--validator-api-port`](#validator-api-port)
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1696]:
```suggestion
* `interface` by using [`--validator-api-interface`](#validator-api-interface).
* `port` by using [`--validator-api-port`](#validator-api-port).
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 11]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    TEKU_VALIDATOR_API_HOST_ALLOWLIST=medomain.com,meotherdomain.com
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-host-allowlist: ["medomain.com", "meotherdomain.com"]
+    ```
+
+A comma-separated list of hostnames to allow access to the validator REST API. By
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1756]:
```suggestion
A comma-separated list of hostnames to allow access to the [validator REST API](../Rest_API/Rest.md#enable-the-validator-client-api). By
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 12]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    TEKU_VALIDATOR_API_INTERFACE=0.0.0.0
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-interface: "0.0.0.0"
+    ```
+
+Specifies the interface on which the validator REST API listens.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1793]:
```suggestion
The interface on which the [validator REST API](../Rest_API/Rest.md#enable-the-validator-client-api) listens.
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 13]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    TEKU_VALIDATOR_API_KEYSTORE_PASSWORD_FILE=keystore_pass.txt
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-keystore-password-file: "keystore_pass.txt"
+    ```
+
+Password used to decrypt the keystore for the validator REST API.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1850]:
```suggestion
Password used to decrypt the keystore for the [validator REST API](../Rest_API/Rest.md#enable-the-validator-client-api).
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 14]
```diff
@@ -1621,6 +1621,263 @@ The default is 5051.
...
+    TEKU_VALIDATOR_API_PORT=5052
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validator-api-port: 5052
+    ```
+
+Specifies Validator REST API listening port (HTTP).
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:1878]:
```suggestion
The [validator REST API](../Rest_API/Rest.md#enable-the-validator-client-api) listening port (HTTP).
```

File: docs/Reference/Rest_API/Rest.md [hunk 15]
```diff
@@ -61,6 +61,22 @@ You can also use tools such as [Postman] or [cURL] to interact with Teku APIs.
...
+[`--validator-api-enabled`](../CLI/CLI-Syntax.md#validator-api-enabled) command line option to `true`.
+
+The [OpenAPI specifications](https://swagger.io/specification/) for the validator client API are available at `/swagger-docs`
+when the [`--validator-api-docs-enabled`](../CLI/CLI-Syntax.md#validator-api-docs-enabled) option is set to `true`.
+The `/swagger-docs` endpoint defines the API if code generators are in use.
+
+When enabling the API documentation endpoint, specify the following:
+
+* `interface` using the [`--validator-api-interface`](../CLI/CLI-Syntax.md#validator-api-interface) option
+* `port` using the [`--validator-api-port`](../CLI/CLI-Syntax.md#validator-api-port) option
```
Comment (alexandratran) [inline docs/Reference/Rest_API/Rest.md:78]:
```suggestion
When enabling the API documentation endpoint, specify:

* `interface` using the [`--validator-api-interface`](../CLI/CLI-Syntax.md#validator-api-interface) option.
* `port` using the [`--validator-api-port`](../CLI/CLI-Syntax.md#validator-api-port) option.
```

**Review by alexandratran — COMMENTED** [review]
Some formatting suggestions.

**Review by alexandratran — APPROVED** [review]
LGTM

**Pull request merged**